### DISCUSSION TITLE
[solved] Power Meter with SML not working

### DISCUSSION THREAD

**Discussion by drucknase** [hoylabs/OpenDTU-OnBattery#533] [Q&A] [answered]
Hello,
my configuration is 1 Hoymiles HM-800 and new a power meter LEPUS from NZR. In the power meter specification for the IR-interface is mentioned 9600Baud, 8N1, protocol SML+COSEM. Up to now I run OpenDTU successfully with the HM-800. Now the power utility installed a power meter LEPUS from NZR and I want to read out the measurements via the integrated IR-interface. So I changed to OpenDTU-onBattery because of the power meter option. I installed a TTL-UART-IR-read head at the power meter. I can measure at GPIO35 at the Nodemcu ESP32 the SML signal with correct bit timing. I also attached a serial analyser to GPIO35 and could captuer a SML frame. This is transmitted by the power meter every 1sec. I enabled power meter and the SML(OBIS 16.7.0) interface in the OpenDTU-onBattery configuration menue.
Unfortunately there is always 0 power measurment on the web GUI and I can also not see any information regarding the power meter on the console output.
In one issue I found a hint to manage the serial interfave in the device manager but found no example and also no documentation for that. So I believe, that softwareserial is 'hardwired' to GPIO35, see line 21 of powermeter.h file. That's why I think it should work without configuration with device manager.
Any ideas how to debug this behaviour? As mentioned, I can measure the SML signal at GPIO35 with oscilloscope and also with serial analyser.
Thank you for your support.

**Comment by drucknase** [comment, marked as answer]
I could solve my problem! There was one parameter in the menue of the powermeter, which was not set correctly. Now it is working.

**Comment by alexseuf** [comment]
so with "menue of the powermeter" you mean within the software/parameter of the meterining device in your meter cabinet or the "power meter" Menue of the "openDTUon battery" firmware. Could you explain what you have changed?
Thaynk you

**Reply by drucknase** [reply]
it was a parameter in the meterining device in the meter cabinet. See attached document for what I did. Unfortunately it is in German.
[NZR_LEPUS_Zweiwegezähler_231126.pdf](https://github.com/helgeerbe/OpenDTU-OnBattery/files/14023716/NZR_LEPUS_Zweiwegezahler_231126.pdf)
